File: samples/get-users-show/ruby-twitter/execute.rb
#!/usr/bin/env ruby

require "rubygems"
require "twitter"

if ARGV.size != 1
  puts "Usage: ./execute.rb SCREEN_NAME"
  puts "Example: ./execute.rb TwitterDev"
  exit
end

# Configuring credentials
credentials = File.read(File.expand_path("../../../credentials.txt")).split("\n");
keys = {
  :consumer_key => credentials[0],
  :consumer_secret => credentials[1],
  :token => credentials[2],
  :token_secret => credentials[3] 
}

client = Twitter::REST::Client.new do |config|
  config.consumer_key        = keys[:consumer_key]
  config.consumer_secret     = keys[:consumer_secret]
  config.access_token        = keys[:token]
  config.access_token_secret = keys[:token_secret]
end

user = client.user(ARGV[0])
puts user.to_h.to_json

File: samples/get-users-show/nodejs-twit/execute.js
#!/usr/bin/env nodejs

var fs = require('fs');
var Twit = require('twit');

// Check if there is a username in the command line
if (!process.argv[2]) {
  console.log('Usage: ./execute.js SCREEN_NAME');
  console.log('Example: ./execute.js TwitterDev');
  process.exit();
}

// Parsing the credential files
var credentials = fs.readFileSync(__dirname + '/../../../credentials.txt', 'utf8').split('\n');

// Creating the client
var client = new Twit({
  consumer_key: credentials[0],
  consumer_secret: credentials[1],
  access_token: credentials[2],
  access_token_secret: credentials[3]
});

// Get the user info
client.get('users/show', {
  screen_name: process.argv[2]
}, function(err, data) {
  if(err)
    throw err;

  console.log(data);
});
